packages/rulesets/src/spectral/functions/resource-must-reference-common-types.ts

```diff
@@ -0,0 +1,57 @@
+// ResourceMustReferenceCommonTypes
+// Validates that resource definitions use the common types TrackedResource or ProxyResource definitions.
+// RPC Code: <not coded yet>
+
+// regex matching paths that end with common-types/resource-management/<version number>/types.json#definitions/<ProxyResource or TrackedResource>
+const RESOURCE_COMMON_TYPES_REGEX = /.*common-types\/resource-management\/v\d+\/types.json#\/definitions\/(Proxy|Tracked)Resource/
+
+export const resourceMustReferenceCommonTypes = (ref: any, _opts: any, ctx: any) => {
+  if (!ref) {
+    return []
+  }
+
+  const swagger = ctx?.documentInventory?.resolved
+  const definitions = swagger?.definitions
+
+  if (!definitions) {
+    return []
+  }
+
+  // check the ref for each 200 response of get, put, patch
+  // e.g.
+  // "responses": {
+  //   "200": {
+  //     "description": "Successfully updated the PrincipalAssignment.",
+  //     "schema": {
+  //       "$ref": "#/definitions/ClusterPrincipalAssignment"
+  //     }
+  //   }
+  // },
```

Comment:
How about we remove this commented out code? If there is a good reason to keep it I would appreciate a comment explaining why.